index.js

```diff
@@ -64,11 +64,8 @@ var uncaughtException = require('./lib/interfaces/uncaught.js');
  */
 function initializeClientAndInterfaces ( initConfiguration ) {
 
-  var config = gatherConfiguration(initConfiguration);
-  var client = new AuthClient(
-    config.projectId,
-    config.shouldReportErrorsToAPI
-  );
+  var config = new Configuration(initConfiguration).init();
```

Comment:
It seems that the init function is supposed to be called immediately after construction and it must be called. The expectation is that it is called only once. Could the `init` function be merged into the constructor, specially if it ends up simplifying the state machine in `Configuration`: `_initState`, `_startedInit`, and `_initError`.